lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -4804,23 +4806,176 @@ LogicalResult StmtEmitter::emitVerifClockedAssertLike(
+                                      op.getEdge(), PPExtString("assert"));
+  }
+  return emitVerifClockedAssertLike(op, property, disableVal, op.getClock(),
+                                    op.getEdge(), PPExtString("assert"));
 }
 
 LogicalResult StmtEmitter::visitVerif(verif::ClockedAssumeOp op) {
-  return emitVerifClockedAssertLike(op, op.getProperty(), op.getDisable(),
-                                    op.getClock(), op.getEdge(),
-                                    PPExtString("assume"));
+  // If the op has a disable, then emit it traditionally
+  if (auto disable = op.getDisable())
+    return emitVerifClockedAssertLike(op, op.getProperty(), disable,
+                                      op.getClock(), op.getEdge(),
+                                      PPExtString("assume"));
+
+  // Try to identify a disable pattern in the property
+  Value disableVal, property;
+  bool disablePattern =
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<comb::OrOp>(mBool(&disableVal), mProperty(&property))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<comb::OrOp>(mProperty(&property), mBool(&disableVal))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<ltl::OrOp>(mProperty(&property), mBool(&disableVal))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<ltl::OrOp>(mBool(&disableVal), mProperty(&property)));
+
+  if (!disablePattern) {
+    mlir::OpBuilder builder(op.getContext());
+    mlir::Value f = builder.createOrFold<hw::ConstantOp>(
+        op.getLoc(), builder.getI1Type(), 0);
+
+    return emitVerifClockedAssertLike(op, op.getProperty(), f, op.getClock(),
+                                      op.getEdge(), PPExtString("assume"));
+  }
+  return emitVerifClockedAssertLike(op, property, disableVal, op.getClock(),
+                                    op.getEdge(), PPExtString("assume"));
 }
 
 LogicalResult StmtEmitter::visitVerif(verif::ClockedCoverOp op) {
-  return emitVerifClockedAssertLike(op, op.getProperty(), op.getDisable(),
-                                    op.getClock(), op.getEdge(),
-                                    PPExtString("cover"));
+  // If the op has a disable, then emit it traditionally
+  if (auto disable = op.getDisable())
+    return emitVerifClockedAssertLike(op, op.getProperty(), disable,
+                                      op.getClock(), op.getEdge(),
+                                      PPExtString("cover"));
+
+  // Try to identify a disable pattern in the property
+  Value disableVal, property, one;
+  bool disablePattern =
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<comb::AndOp>(
+              mProperty(&property),
+              mlir::m_Op<comb::XorOp>(mBool(&disableVal), mOne(&one)))) ||
+      // Not sure if the pattern matchers are naturally commutative, so just to
```

Comment:
Can we define "canonical order" of operands for `ltl::AndOp`? For example we can create canonicalizer  that moves every ltl.property operand to the head. This logic is similar to comb::XOrOp::isBinaryNot(https://github.com/llvm/circt/blob/76b65f8606de447a85ef63c9c98ae9b1e2bd4b7f/lib/Dialect/Comb/CombOps.cpp#L224-L231). MLIR canonicalizer pushes constant operands to the tail for a commutative operation.